numberformat/proposed.emu

```diff
@@ -230,16 +230,20 @@
 
       <emu-alg>
         1. Let _exponent_ be 0.
-        1. If _x_ is *NaN*, then
+        1. If _x_ is ~not-a-number~, then
           1. Let _n_ be an implementation- and locale-dependent (ILD) String value indicating the *NaN* value.
-        1. Else if _x_ is *+&infin;*, then
+        1. Else if _x_ is ~positive-infinity~, then
           1. Let _n_ be an ILD String value indicating positive infinity.
-        1. Else if _x_ is *-&infin;*, then
+        1. Else if _x_ is ~negative-infinity~, then
           1. Let _n_ be an ILD String value indicating negative infinity.
         1. Else,
-          1. If _numberFormat_.[[Style]] is *"percent"*, let _x_ be 100 × _x_.
-          1. Let _exponent_ be ComputeExponent(_numberFormat_, _x_).
-          1. Let _x_ be _x_ × 10<sup>-_exponent_</sup>.
+          1. If _x_ is ~negative-zero~,
+            1. Let _x_ be *-0*.
```

Comment:
I think it is much more confusing to have two "versions" of -0 (the enumerated type and the float type) than to keep with a single enumerated type.

Reply:
Ah, I wasn't considering the incongruity (maybe I was over-focused on the branches). We might resolve that tension by pushing this change toward labels into the signature of FormatNumericToString. Currently, it accepts "a Mathematical value or **-0**", but we can eliminate this new branch if we change it to "a Mathematical value or `~negative-zero~`. I've pushed up a commit to this effect--what do you think?